sonar-text-dotnet/pipeline_scripts/build_and_pack_solution.ps1

```diff
@@ -0,0 +1,43 @@
+function CheckIfSuccessful($StepName) {
+if(!$?) {
+    Write-Host "sonar-text-dotnet ${StepName} failed."
+    Exit $LASTEXITCODE
+}
+}
+
+Set-Location $env:PROJECT_DIR
+$SIGN_DOTNET_ASSEMBLY = "$CIRRUS_BRANCH" -eq "master" -or "$CIRRUS_BRANCH".startsWith("branch-.*")
+Write-Host "Should the assembly be signed: ${SIGN_DOTNET_ASSEMBLY}"
+
+Write-Host Restoring $env:SOLUTION_DIR
+$nugetFeed = "https://pkgs.dev.azure.com/sonarsource/399fb241-ecc7-4802-8697-dcdd01fbb832/_packaging/slvs_input/nuget/v3/index.json"
+# For nuget version see: https://github.com/SonarSource/re-ci-images/blob/master/ec2-images/build-base-windows-dotnet.pkr.hcl#L39
+nuget restore -LockedMode -Source $nugetFeed
+CheckIfSuccessful "nuget restore"
+
+Write-Host Building $env:SOLUTION_DIR
+dotnet build `
+    $env:SOLUTION_DIR `
+    --no-restore `
+    /nologo `
+    /nr:false `
+    /p:platform="Any CPU" `
+    /p:configuration=$env:BUILD_CONFIGURATION `
+    /p:VisualStudioVersion="17.0" `
+    /p:CommitId=$env:CIRRUS_CHANGE_IN_REPO `
+    /p:BranchName=$env:CIRRUS_BRANCH `
+    /p:BuildNumber=$env:BUILD_NUMBER `
+    /p:SignAssembly=$SIGN_DOTNET_ASSEMBLY `
+    /p:AssemblyOriginatorKeyFile=$env:SNK_PATH
+CheckIfSuccessful "build"
+
+Write-Host Packing $env:SOLUTION_DIR
+dotnet pack $env:SOLUTION_DIR -o $env:PROJECT_DIR\artifacts -c $env:BUILD_CONFIGURATION --no-build
+CheckIfSuccessful "packaging"
+
+if($SIGN_DOTNET_ASSEMBLY) {
```

Comment:
It is the same as in the previous pipeline. And the ITs are run both on master and PR, and they will be green and inconclusive on non-master/feature branch runs.